internal/controller/managedcluster_controller.go

```diff
@@ -337,6 +338,42 @@ func (r *ManagedClusterReconciler) Update(ctx context.Context, managedCluster *h
 	return ctrl.Result{}, nil
 }
 
+func (r *ManagedClusterReconciler) needToRequeue(ctx context.Context, managedCluster *hmc.ManagedCluster) (bool, error) {
+	type objectToCheck struct {
+		gvr       schema.GroupVersionResource
+		condition string
+	}
+
+	var requeue bool
+	var errs error
+	for _, obj := range []objectToCheck{
+		{
+			gvr: schema.GroupVersionResource{
+				Group:    "cluster.x-k8s.io",
+				Version:  "v1beta1",
+				Resource: "clusters",
+			},
+			condition: "ControlPlaneInitialized",
+		},
+		{
+			gvr: schema.GroupVersionResource{
+				Group:    "cluster.x-k8s.io",
+				Version:  "v1beta1",
+				Resource: "machinedeployments",
+			},
+			condition: "Available",
+		},
+	} {
+		r, err := r.setStatusFromChildObjects(ctx, managedCluster, obj.gvr, obj.condition)
+		errs = errors.Join(errs, err)
+		if r {
```

Comment:
please address the comments re supporting 0 and multiple machinedeployments:
* don't error or retry if 0, process all if > 1

Reply:
>when do we have several MachineDeployments?
when you have different requirements for nodes. It's not in our current templates, but should be supported in the framework